encryptor/network/message.py

```diff
@@ -146,30 +150,28 @@ def from_message(message: Message) -> "JSONMessageContent":
         content_type = message.headers.content_type
 
         if content_type != ContentType.JSON:
-            raise ValueError(
+            raise TypeError(
                 f"Cannot extract JSON content from a message with declared content type as {content_type}"
             )
 
-        data = _decode_json(message.content, JSONMessageContent.ENCODING)
+        data = json_decode(message.content.decode(JSONMessageContent.ENCODING))
 
         if not isinstance(data, dict):
-            raise ValueError(
+            raise TypeError(
                 f"Cannot convert {data!r} to JSON content, value is not a dictionary"
             )
 
         for field in JSONMessageContent.REQUIRED_FIELDS:
             if field not in data:
                 raise ValueError(f"Missing a required field {field}")
 
-        data["content_type"] = JSONContentType(data["content_type"])
-
         return JSONMessageContent(**data)
 
 
 class MessageReader:
     """Reads messages from an endpoint."""
 
-    def __init__(self, sock: socket.socket) -> None:
+    def __init__(self, sock: socket.socket, keys_dir: str) -> None:
```

Comment:
Remove `keys_dir` from the server thread.

Reply:
done